sandy/core/cov.py

```diff
@@ -258,12 +259,6 @@ def data(self):
     @data.setter
     def data(self, data):
         self._data = data.astype(float)

```

Comment:
Maybe it should be `self._data = pd.DataFrame(data, dtype=float)`.
Then your input can also be a numpy matrix.